src/GraphQL/Types/GraphTypesLookup.cs

```diff
@@ -22,41 +72,24 @@ public GraphTypesLookup(INameConverter nameConverter)
             if (nameConverter == null)
                 throw new ArgumentNullException(nameof(nameConverter));
 
-            // standard scalars https://graphql.github.io/graphql-spec/June2018/#sec-Scalars
-            AddType<StringGraphType>();
-            AddType<BooleanGraphType>();
-            AddType<FloatGraphType>();
-            AddType<IntGraphType>();
-            AddType<IdGraphType>();
-
-            // .NET custom scalars
-            AddType<DateGraphType>();
-            AddType<DateTimeGraphType>();
-            AddType<DateTimeOffsetGraphType>();
-            AddType<TimeSpanSecondsGraphType>();
-            AddType<TimeSpanMillisecondsGraphType>();
-            AddType<DecimalGraphType>();
-            AddType<UriGraphType>();
-            AddType<GuidGraphType>();
-            AddType<ShortGraphType>();
-            AddType<UShortGraphType>();
-            AddType<UIntGraphType>();
-            AddType<LongGraphType>();
-            AddType<BigIntGraphType>();
-            AddType<ULongGraphType>();
-            AddType<ByteGraphType>();
-            AddType<SByteGraphType>();
-
-            // Add introspection types.  Note that introspection types rely on the
-            //   CamelCaseNameConverter, as some fields are defined in pascal case - e.g. Field(x => x.Name)
+            _context = new TypeCollectionContext(
```

Comment:
`TypeCollectionContext` is just like a proxy to resolve/add methods pair. It can be shared across all introspection types whilst `AddType` calls.